### PR TITLE
fix: [M3-8685] - Users unable to update timezone

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

- Fixes bug preventing users from updating / setting their timezone 🌎 🕐 
- I think this might be a regression from https://github.com/linode/manager/pull/10780

## Preview 📷

| Before  | After   |
| ------- | ------- |
| <video src="https://github.com/user-attachments/assets/f5c3392c-138b-4f02-aa4b-b255a9e9bd58" /> | <video src="https://github.com/user-attachments/assets/24a84f42-9c5e-4899-b8eb-a722de6f00af" /> |

## How to test 🧪

- Checkout this PR or use the preview link
- Verify you can update your timezone on http://localhost:3000/profile/display

## As an Author I have considered 🤔

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [ ] ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/features/Profile/DisplaySettings/TimezoneForm.tsx [hunk 1]
```diff
@@ -3,160 +3,111 @@ import { DateTime } from 'luxon';
...
+  const [timezoneValue, setTimezoneValue] = React.useState(profile?.timezone);
 
   const onSubmit = () => {
-    if (timezoneValue === '') {
-      return;
+    if (!timezoneValue) {
+      enqueueSnackbar('Please select a valid timezone.', { variant: 'error' });
     }
 
-    updateProfile({ timezone: String(timezoneValue) }).then(() => {
```
Comment (bnussman-akamai) [inline packages/manager/src/features/Profile/DisplaySettings/TimezoneForm.tsx:71]: The is the cause of the bug. `timezoneValue` is an object and we "cast" it to a string, causing us to send `{"timezone": "[Object object]"}`  to the API

File: packages/manager/src/features/Profile/DisplaySettings/TimezoneForm.tsx [hunk 2]
```diff
@@ -3,160 +3,111 @@ import { DateTime } from 'luxon';
...
-const renderTimezonesList = (): TimezoneOption<string>[] => {
+const getTimezoneOptions = () => {
   return timezones
-    .map((tz) => ({ ...tz, offset: DateTime.now().setZone(tz.name).offset }))
-    .sort((a, b) => a.offset - b.offset)
-    .map((tz: Timezone) => {
-      const label = formatOffset(tz);
-      return { label, value: tz.name };
-    });
+    .map((tz) => {
```
Comment (bnussman-akamai) [inline packages/manager/src/features/Profile/DisplaySettings/TimezoneForm.tsx:32]: I was able to consolidate the two `map`s into one

File: packages/manager/src/features/Profile/DisplaySettings/TimezoneForm.tsx [hunk 3]
```diff
@@ -3,160 +3,111 @@ import { DateTime } from 'luxon';
...
+export const getOptionLabel = ({ label, offset }: Timezone) => {
   const minutes = (Math.abs(offset) % 60).toLocaleString(undefined, {
     minimumIntegerDigits: 2,
     useGrouping: false,
   });
   const hours = Math.floor(Math.abs(offset) / 60);
   const isPositive = Math.abs(offset) === offset ? '+' : '-';
 
-  return `\(GMT ${isPositive}${hours}:${minutes}\) ${label}`;
+  return `(GMT ${isPositive}${hours}:${minutes}) ${label}`;
```
Comment (bnussman-akamai) [inline packages/manager/src/features/Profile/DisplaySettings/TimezoneForm.tsx:27]: I don't think the `\`s are necessary. I removed them

File: packages/manager/src/features/Profile/DisplaySettings/TimezoneForm.tsx [hunk 4]
```diff
@@ -3,160 +3,111 @@ import { DateTime } from 'luxon';
...
-interface Timezone {
-  label: string;
-  name: string;
-  offset: number;
-}
-
-export interface TimezoneOption<T = string, L = string> {
-  label: L;
-  value: T;
-}
```
Comment (bnussman-akamai) [inline packages/manager/src/features/Profile/DisplaySettings/TimezoneForm.tsx:27]: We don't need these extra types. We can lean on TypeScript's inference to get a solution that is even more typesafe!

File: packages/manager/src/features/Profile/DisplaySettings/TimezoneForm.tsx [hunk 5]
```diff
@@ -3,160 +3,111 @@ import { DateTime } from 'luxon';
...
+  const disabled = !timezoneValue || profile?.timezone === timezoneValue;
 
   if (!profile) {
     return <CircleProgress />;
   }
 
   return (
     <>
-      {loggedInAsCustomer ? (
-        <StyledLoggedInAsCustomerNotice data-testid="admin-notice">
```
Comment (bnussman-akamai) [inline packages/manager/src/features/Profile/DisplaySettings/TimezoneForm.tsx:90]: Removed this one-off component in favor of a simple `Notice` just to clean things up a bit

File: packages/manager/src/features/Profile/DisplaySettings/TimezoneForm.tsx [hunk 6]
```diff
@@ -3,160 +3,111 @@ import { DateTime } from 'luxon';
...
-          <Typography variant="h2">
-            While you are logged in as a customer, all times, dates, and graphs
-            will be displayed in your browser&rsquo;s timezone ({timezone}).
```
Comment (bnussman-akamai) [inline packages/manager/src/features/Profile/DisplaySettings/TimezoneForm.tsx:93]: This copy was just factually incorrect. 

 "all times, dates, and graphs will be displayed in your browser's timezone" has been corrected to say "all times, dates, and graphs will be displayed in the user's timezone "

File: packages/manager/src/features/Profile/DisplaySettings/TimezoneForm.tsx [hunk 7]
```diff
@@ -3,160 +3,111 @@ import { DateTime } from 'luxon';
...
-        <Stack>
-          <Autocomplete
-            slotProps={{
-              popper: {
-                sx: {
-                  maxHeight: '285px',
-                  overflow: 'hidden',
-                },
-              },
-            }}
```
Comment (bnussman-akamai) [inline packages/manager/src/features/Profile/DisplaySettings/TimezoneForm.tsx:107]: I was able to clean up a lot of these one-off styles by using the `fullWidth` prop instead

File: packages/manager/src/features/Profile/DisplaySettings/TimezoneForm.tsx [hunk 8]
```diff
@@ -3,160 +3,111 @@ import { DateTime } from 'luxon';
...
-            disableClearable
-            errorText={error?.[0].reason}
-            label="Timezone"
-            onChange={(_, option) => handleTimezoneChange(option)}
-            options={timezoneList}
-            placeholder="Choose a Timezone"
-            sx={{ width: '416px' }}
-          />
-        </Stack>
-        <ActionsPanel
```
Comment (bnussman-akamai) [inline packages/manager/src/features/Profile/DisplaySettings/TimezoneForm.tsx:123]: Replaced the `ActionsPanel` with a `Button`. I think a plain Button is more fitting here.

File: packages/manager/src/features/Profile/DisplaySettings/TimezoneForm.tsx [hunk 9]
```diff
@@ -3,160 +3,111 @@ import { DateTime } from 'luxon';
...
+          Update Timezone
+        </Button>
+      </TimezoneFormContainer>
     </>
   );
 };
 
-const StyledRootContainer = styled('div', {
-  label: 'StyledRootContainer',
+const TimezoneFormContainer = styled('div', {
```
Comment (abailly-akamai) [inline packages/manager/src/features/Profile/DisplaySettings/TimezoneForm.tsx:102]: It's a nit, but I think we should continue to name Styled Components with "Styled", it makes it so much easier for developers to quickly identify components that are purely meant to apply styles and and hove other function.

Doesn't need to be changed for this PR but something to consider.

**Review by abailly-akamai — DISMISSED** [review]
Thanks!

Confirmed the bug ad the resolution ✅ 
Other changes make sense to me and do not add any regression on my end

**Comment by abailly-akamai** [comment]
@bnussman-akamai the e2e failure appears to be relevant here

Screenshot 2024-10-02 at 09 25 10

File: packages/manager/src/features/Profile/DisplaySettings/TimezoneForm.tsx (hunk 9 above)
Comment (bnussman-akamai) [reply packages/manager/src/features/Profile/DisplaySettings/TimezoneForm.tsx:102]: Ahh got it. I always though that prefixing the component with "Styled" was kind of an anti-pattern but I see what you mean

**Review by cpathipa — DISMISSED** [review]
Confirming on the bug timezone is successfully updating.

**Comment by bnussman-akamai** [comment]
Good catch @abailly-akamai 

Should be fixed by d7ce808

I removed the `autoHighlight` prop on the Autocomplete in attempt to clean things up, which caused it to behave differently (caused it to not automatically select the first option). I re-added that, which should fix the test.

**Review by coliu-akamai — DISMISSED** [review]
✅ confirmed can now update my timezone
✅ confirmed previously failing e2e passing locally now

thanks for the fix + cleanup! 🧹

**Review by coliu-akamai — APPROVED** [review]

**Review by jdamore-linode — APPROVED** [review]
Thanks for the catch and fix, @bnussman-akamai!

**Review by abailly-akamai — APPROVED** [review]
Thanks! Approving pending e2e passing

**Review by hana-linode — APPROVED** [review]

**Pull request merged**